target/linux/lantiq/files-5.4/arch/mips/boot/dts/lantiq/vr9.dtsi

```diff
@@ -429,22 +413,73 @@
 			phy-names = "usb2-phy";
 		};
 
-		eth0: eth@e108000 {
+		gswip: switch@e108000 {
+			compatible = "lantiq,xrx200-gswip";
 			#address-cells = <1>;
 			#size-cells = <0>;
-			compatible = "lantiq,xrx200-net";
-			reg = <	0xe108000 0x3000 /* switch */
-				0xe10b100 0x70 /* mdio */
-				0xe10b1d8 0x30 /* mii */
-				0xe10b308 0x30 /* pmac */
-			>;
-			interrupt-parent = <&icu0>;
-			interrupts = <75 73 72>;
-			resets = <&reset0 21 16>, <&reset0 8 8>;
-			reset-names = "switch", "ppe";
-			lantiq,phys = <&gphy0>, <&gphy1>;
+			reg = < 0xe108000 0x3000 /* switch */
+				0xe10b100 0x70   /* mdio */
+				0xe10b1d8 0x30   /* mii */
+				>;
 			pinctrl-0 = <&mdio_pins>;
 			pinctrl-names = "default";
+
+			dsa,member = <0 0>;
+
+			gswip_ports: ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@6 {
+					reg = <0x6>;
+					label = "cpu";
+					ethernet = <&eth0>;
+				};
+			};
+
+			gswip_mdio: mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "lantiq,xrx200-mdio";
+			};
+
+			gphy-fw {
+				compatible = "lantiq,xrx200-gphy-fw", "lantiq,gphy-fw";
+				lantiq,rcu = <&rcu0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				gphy0: gphy@20 {
+					reg = <0x20>;
+
+					resets = <&reset0 31 30>;
```

Comment:
gphy2 reset was removed. Is this intentional?

Reply:
good spot - thank you!
I don't remember doing this on purpose.
@hauke can you confirm the second pair of resets is still needed here (Documentation/devicetree/bindings/net/dsa/lantiq-gswip.txt is not clear about it either)?